By default, glibc 2.4 installs a gnu/stubs.h which includes either
gnu/stubs-32.h or gnu/stubs-64.h based on the value of __WORDSIZE.
However, only gnu/stubs-32.h is installed, gnu/stubs-64.h is missing on
32 bit platforms. This causes compilation failure in perl 5.8.7 when it
tries to precompile the libc headers.

diff -Naur glibc-2.4.orig/Makefile glibc-2.4/Makefile
--- glibc-2.4.orig/Makefile	2006-03-01 11:35:45.000000000 +0100
+++ glibc-2.4/Makefile	2006-03-08 17:21:12.000000000 +0100
@@ -158,7 +158,7 @@
 
 subdir-stubs := $(foreach dir,$(subdirs),$(common-objpfx)$(dir)/stubs)
 
-ifeq ($(biarch),no)
+ifeq (yes,yes)
 installed-stubs = $(inst_includedir)/gnu/stubs.h
 else
 installed-stubs = $(inst_includedir)/gnu/stubs-$(biarch).h
